service/blocksyncer/block_syncer.go

```diff
@@ -35,27 +42,94 @@ var (
 	eventMap *sync.Map
 	txMap    *sync.Map
 
-	LatestBlockHeight atomic.Value
-	CatchUpFlag       atomic.Value
+	MainService   *BlockSyncer
+	BackupService *BlockSyncer
+
+	FlagDB database.Database
+
+	NeedBackup bool
+
+	CancelMain func()
+	CtxMain    context.Context
 )
```

Comment:
Maybe in the future it can be encapsulated inside the module and switched internally.